numpy/conftest.py

```diff
@@ -133,13 +133,13 @@ def check_fpu_mode(request):
     new_mode = get_fpu_mode()
 
     if old_mode != new_mode:
-        raise AssertionError("FPU precision mode changed from {0:#x} to {1:#x}"
+        raise AssertionError("FPU precision mode changed from {:#x} to {:#x}"
                              " during the test".format(old_mode, new_mode))
```

Comment:
Maybe make this into an f-string?

Reply:
The ruff linter applies these changes automatically.

Rules [UP031](https://docs.astral.sh/ruff/rules/printf-string-formatting/) and [UP032](https://docs.astral.sh/ruff/rules/f-string/) are about change classic `%` interpolation into `format()` and `format()` into f-strings. I have not applied these rules because they would result in extensive changes. I'd rather defer to a specific PR.